client/app/pages/dashboards/dashboard.js

```diff
@@ -428,15 +428,14 @@ function DashboardCtrl(
   }
 
   this.openShareForm = () => {
-    // check if any of the wigets have query parameters
-    const hasQueryParams = _.some(
+    const hasOnlySafeQueries = _.every(
       this.dashboard.widgets,
-      w => !_.isEmpty(w.getQuery() && w.getQuery().getParametersDefs()),
+      w => w.getQuery() && w.getQuery().is_safe,
```

Comment:
when no param in any query, this will return false, then `hasOnlySafeQueries` would return false.
Change to below would work:
`w => !w.getQuery() || w.getQuery().is_safe,`

Reply:
I believe `is_safe` should return `true` for queries without any param (at least this happens in the backend - I'll have to verify that this is the same in the frontend)

Reply:
I hardcoded `is_safe` in `ParameterizedQuery` to `True`, still not working. After changing above mentioned code, it works now.
That's what I did in my redash instance.

Reply:
Besides, `is_safe` in `ParameterizedQuery` could be simplified as follow:
```
return all(i['type'] != 'text' for i in self.schema)
```

Reply:
> I believe `is_safe` should return `true` for queries without any param (at least this happens in the backend - I'll have to verify that this is the same in the frontend)

IMHO It's not about `is_safe`, it's about `getQuery()` may return null as the widget does not contain any param.

Reply:
@truebit you're right, I didn't interpret your comment correctly. This is fixed in c00c720. Thanks @arikfr for clarifying.